qujax/observable.py

```diff
@@ -4,52 +4,62 @@
+        hermitian: Hermitian array
+            must be in tensor form with shape (2,2,...).
+        qubit_inds: Sequence of qubit indices for Hermitian to be applied to.
+            Must have 2 * len(qubit_inds) = hermitian.ndim
 
     Returns:
-        Function that takes statetensor and returns expected value (float).
+        Expected value (float).
     """
+    statetensor_new = apply_gate(statetensor, hermitian, qubit_inds)
+    axes = tuple(range(statetensor.ndim))
+    return jnp.tensordot(statetensor.conjugate(), statetensor_new, axes=(axes, axes)).real
 
-    def statetensor_to_single_expectation(statetensor: jnp.ndarray) -> float:
-        """
-        Evaluates expected value of statetensor through gate.
 
-        Args:
-            statetensor: Input statetensor.
+def check_hermitian(hermitian: Union[str, jnp.ndarray]):
+    """
+    Checks whether a matrix or tensor is Hermitian.
 
-        Returns:
-            Expected value (float).
-        """
-        statetensor_new = jnp.tensordot(gate_tensor, statetensor,
-                                        axes=(list(range(-len(qubit_inds), 0)), qubit_inds))
-        statetensor_new = jnp.moveaxis(statetensor_new, list(range(len(qubit_inds))), qubit_inds)
-        axes = tuple(range(statetensor.ndim))
-        return jnp.tensordot(statetensor.conjugate(), statetensor_new, axes=(axes, axes)).real
+    Args:
+        hermitian: array containing potentially Hermitian matrix or tensor
 
-    return statetensor_to_single_expectation
+    """
+    if isinstance(hermitian, str):
+        if hermitian not in paulis:
+            raise TypeError(f'qujax only accepts {tuple(paulis.keys())} as Hermitian strings, received: {hermitian}')
+    else:
+        n_qubits = hermitian.ndim // 2
+        hermitian_mat = hermitian.reshape(2 * n_qubits, 2 * n_qubits)
+        if not jnp.allclose(hermitian_mat, hermitian_mat.T.conj()):
+            raise TypeError(f'Array not Hermitian: {hermitian}')
 
 
-def get_statetensor_to_expectation_func(gate_seq_seq: Sequence[Sequence[Union[str, jnp.ndarray]]],
+def get_statetensor_to_expectation_func(hermitian_seq_seq: Sequence[Sequence[Union[str, jnp.ndarray]]],
                                         qubits_seq_seq: Sequence[Sequence[int]],
                                         coefficients: Union[Sequence[float], jnp.ndarray]) \
         -> Callable[[jnp.ndarray], float]:
     """
-    Converts gate strings (or arrays), qubit indices and coefficients into a function that
-    converts statetensor into expected value.
+    Converts strings (or arrays) representing Hermitian matrices, qubit indices and
+    coefficients into a function that converts a statetensor into an expected value.
 
     Args:
-        gate_seq_seq: Sequence of sequences of gates.
-            Each gate is either a tensor (jnp.ndarray) or a string corresponding to an array in qujax.gates.
+        hermitian_seq_seq: Sequence of sequences of Hermitian matrices/tensors.
```

Comment:
```suggestion
        hermitian_seq_seq: Sequence of sequences of Hermitian matrices (in tensor form).
```

Reply:
here they don't actually have to be in tensor form as they will be reshaped anyway